plugins/org.yakindu.sct.model.stext/src/org/yakindu/sct/model/stext/validation/STextValidationMessages.java

```diff
@@ -61,4 +61,6 @@
 	public static final String CONTRADICTORY_ANNOTATIONS = "Some annotations (%s) have contradictory effects.";
 	public static final String BAD_EVENT_NAMES = "'%s' is already used as name of an internal event.";
 	public static final String SYNC_OUTGOING_TRIGGER = "Triggers and guards on a synchronization's outgoing transitions will be ignored.";
+	public static final String ALWAYS_TRUE_TRANSITION_USED = "Transition with '%s' should have the lowest priority, otherwise transitions with lower priority will never be taken.";
+	public static final String USE_DEFAULT_TRIGGER_IN_CHOICES = "Transition with '%s' effect blocking other transitions. Use 'default' or 'else' instead.";
```

Comment:
The problem is nearly the same as for the other check. So use a comparable message like: 

"Transition with '%s' should be evaluate as the last transition of the state as following transitions won't be evaluated. Use 'default' or 'else' instead or change the transition order (in the Properties view)."

Be aware that the use of 'always' and 'default' are conflicting if both are defined. In this case the 'default' transition will never be taken. This is an additional case that has to be considered.

Reply:
This are two different properties that we have to check. 
1. There should be no dead transitions.
2. It must always be possible to exit a choice.

Taking the existing semantics we can identify the following cases:
a. Using always not with lowest priority violates 1). 
b. Using always with lowest priotity and there is no additional default fulfills 1) and 2)
c. Using always and default together will always violate 2) at least for the default transition.
d. Using two or more default transition also leads to dead transitions (violates 2))

I derive the following conclusions from this: 

i)  a) is exactly the same problem as for regular states. An same problem should be reported with the same message/reason. An additional possible fix in the context of choices is to convert always to default or to fix the order as for regular states.

ii)  Converting always to default is invalid if there is already a default transition - always and default are always conflicting.

iii) if b) is fulfilled than the warning that you should use a choice should not appear. But currently it will appear.

iv) There must be an additional check for multiple defaults.

Reply:
@rherrmannr: please fix iii)